ui/jobs/components/mch.js

```diff
@@ -41,105 +41,86 @@ export function setup(bars) {
-    refreshWildFireGuage();
-  });
-
   const drillBox = bars.addProcBox({
     id: 'mch-procs-drill',
     fgColor: 'mch-color-drill',
   });
-
   bars.onUseAbility([
     kAbility.Drill,
     kAbility.Bioblaster,
   ], () => {
     drillBox.duration = 0;
     drillBox.duration = calcGCDFromStat(bars, bars.skillSpeed, 20000);
-    refreshWildFireGuage();
   });
 
   const airAnchorBox = bars.addProcBox({
     id: 'mch-procs-airanchor',
     fgColor: 'mch-color-airanchor',
   });
-
   bars.onUseAbility([
     kAbility.AirAnchor,
     kAbility.HotShot,
   ], () => {
     airAnchorBox.duration = 0;
     airAnchorBox.duration = calcGCDFromStat(bars, bars.skillSpeed, 40000);
-    refreshWildFireGuage();
   });
 
+  // Wild Fire Gauge
+  const stacksContainer = document.createElement('div');
+  stacksContainer.id = 'mch-stacks';
+  stacksContainer.classList.add('hide');
+  bars.addJobBarContainer().appendChild(stacksContainer);
+  const wildFireContainer = document.createElement('div');
+  wildFireContainer.id = 'mch-stacks-wildfire';
+  stacksContainer.appendChild(wildFireContainer);
+  const wildFireStacks = [];
+  for (let i = 0; i < 6; ++i) {
+    const d = document.createElement('div');
+    wildFireContainer.appendChild(d);
+    wildFireStacks.push(d);
+  }
+
+  let wildFireCounts = 0;
+  let wildFireActive = false;
+  const refreshWildFireGuage = () => {
+    for (let i = 0; i < 6; ++i) {
+      wildFireStacks[i].classList.remove('fix', 'active');
+      if (wildFireCounts > i) {
+        if (wildFireActive)
+          wildFireStacks[i].classList.add('active');
+        else
+          wildFireStacks[i].classList.add('fix');
+      }
+    }
+  };
+  bars.onMobGainsEffectFromYou(EffectId.Wildfire, (id, e) => {
+    wildFireActive = true;
+    wildFireCounts = e.count;
+    refreshWildFireGuage();
+  });
+  bars.onMobLosesEffectFromYou(EffectId.Wildfire, () => {
+    wildFireActive = false;
+    refreshWildFireGuage();
+  });
   const wildFireBox = bars.addProcBox({
     id: 'mch-procs-wildfire',
     fgColor: 'mch-color-wildfire',
   });
   bars.onUseAbility(kAbility.WildFire, () => {
     wildFireBox.duration = 0;
-    wildFireBox.duration = 10;
+    wildFireBox.duration = 10 + 0.9; // animation delay
     wildFireBox.threshold = 1000;
-    wildFireActive = true;
-    wildFireGCD = -1;
-    refreshWildFireGuage();
     stacksContainer.classList.remove('hide');
     wildFireBox.fg = computeBackgroundColorFrom(wildFireBox, 'mch-color-wildfire.active');
     setTimeout(() => {
-      wildFireBox.duration = 110;
+      wildFireBox.duration = 110 - 0.9;
       wildFireBox.threshold = bars.gcdSkill + 1;
-      wildFireActive = false;
-      wildFireGCD = -1;
-      refreshWildFireGuage();
-      stacksContainer.classList.add('hide');
       wildFireBox.fg = computeBackgroundColorFrom(wildFireBox, 'mch-color-wildfire');
     }, 10000);
+    setTimeout(() => {
+      stacksContainer.classList.add('hide');
+      wildFireCounts = 0;
+    }, 12500);
```

Comment:
I think this could be longer if you wanted (5s?), but I think 2.5s is also fine if you wanted to leave it as-is.